src/reflect/scala/reflect/internal/TreeInfo.scala

```diff
@@ -1078,4 +1078,16 @@ trait MacroAnnotionTreeInfo { self: TreeInfo =>
       SyntacticClassDef(mods, name, tparams, constrMods, vparamss.map(_.map(_.duplicate)), earlyDefs, parents, selfType, body)
     }
   }
+
+  // Return a pair consisting of (all statements up to and including superclass and trait constr calls, rest)
+  final def splitAtSuper(stats: List[Tree], classOnly: Boolean): (List[Tree], List[Tree]) = {
+    def isConstr(tree: Tree): Boolean = tree match {
+      case Block(_, expr) => isConstr(expr) // scala/bug#6481 account for named argument blocks
+      case _              => (tree.symbol ne null) && (if (classOnly) tree.symbol.isClassConstructor else tree.symbol.isConstructor)
+    }
+    val (pre, rest0)       = stats span (!isConstr(_))
+    val (supercalls, rest) = rest0 span (isConstr(_))
+    (pre ::: supercalls, rest)
```

Comment:
This seems like a lot of iteration and a lot of allocation. Would this not be possible to solve with a custom Ordering + a sort?

Reply:
Yes it could be done with a bit less garbage, but this isn't a hotspot based on profiles I've collected.

Reply:
It's your call :) I always err on the side of efficiency :)